src/utils/tasks.js

```diff
@@ -118,16 +125,37 @@ function formatDuration (dur, allowZeros = false) {
   return undefined
 }
 
-function dtMean (taskNode) {
+export function dtMean (taskNode) {
   // Convert to an easily read duration format:
   const dur = taskNode.node?.task?.meanElapsedTime
   return formatDuration(dur)
 }
 
-export {
-  extractGroupState,
-  latestJob,
-  jobMessageOutputs,
-  formatDuration,
-  dtMean
+/**
+ * @param {string} flowNums - Flow numbers in DB format
+ * @returns {string} - Flow numbers in pretty format
+ */
+export function formatFlowNums (flowNums) {
+  return JSON.parse(flowNums).join(', ') || 'None'
+}
+
+/**
+ * Return whether a task has concrete flow numbers
+ * (i.e. not data store "ghost tasks").
+ *
+ * @param {TaskNode} node
+ * @returns {boolean}
+ */
+export function flowNumsValid (node) {
+  return Boolean(node.flowNums && (node.state !== 'waiting' || node.isQueued))
```

Comment:
This is a little hacky and I wonder if it would make more sense to avoid including flow numbers in the cylc-flow data store for tasks that are not yet in the pool?

Reply:
In principle I think that future tasks (that have never run in any flow) should not have any flow numbers.

Reply:
We could, potentially, make this change to cylc-flow, however, we would still require this code in order to support workflows running with older versions of cylc-flow (or accept different behaviour for different versions).

Reply:
I am sticking with the simplest choice for now: show these flow numbers anyway.